paddle/framework/framework.proto

```diff
@@ -118,6 +118,14 @@ message LoDTensorArrayDesc {
 
 message ReaderDesc { repeated LoDTensorDesc lod_tensor = 1; }
 
+message ChanEleDesc { repeated VarDesc meta_data = 1; }
+
+message ChanDesc {
+  repeated ChanEleDesc channel_type = 1;
```

Comment:
I think `ChanEleDesc` maybe not needed, each element inside one channel instance should be of the same type. So the `ChanDesc` should look like: 

```protobuf
message ChanDesc {
required VarType ele_type = 1;
required int32 cap = 2 [ default = 0 ];
required string name = 3;
}
```

Then `VarType` should have a new type called `VAR_LIST` representing a list of variables, so that element like `tensor, label` is able to put inside the channel.

The contents of the channel are manipulated at runtime.

Reply:
@typhoonzero  thanks for your review! And this is a good suggestion.

Reply:
I think all the work to do in this PR is to add the following definition of `ChannelDesc`:

```protobuf
message ChannelDesc {
  VarDesc elem_type = 1;
  int cap = 2;
}
```

Nothing more than that.